extensions/ipynb/src/notebookAttachmentCleaner.ts

```diff
@@ -0,0 +1,250 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+class AttachmentCleaner {
+
+	// Format:       { notebookUri : { cellFragment : { filename : { mime : b64 } } } }
+	attachmentCache: { [key: string]: any } = {};
```

Comment:
You should be able to add better typings for this, similar to what the comment above shows